cmake/Boost.cmake

```diff
@@ -0,0 +1,52 @@
+#option(USE_PYTHON "enable python support" FALSE)
+#option(USE_PYTHON2 "if USE_PYTHON enabled, use python2 instead of python3" FALSE)
+
+set (EP_BOOST "boost")
+
+
+set(BOOST_BUILD_COMMAND ./b2 install --layout=tagged --prefix=${CMAKE_INSTALL_PREFIX} --build-dir=${CMAKE_CURRENT_BINARY_DIR}/build -j${BUILD_JOBS} address-model=64 link=shared runtime-link=shared threading=multi variant=release --with-atomic --with-program_options --with-regex --with-date_time --with-system --with-thread --with-iostreams --with-filesystem --with-serialization --with-wave --with-chrono)
+
+set(BOOST_PYTHON_VERSIONS)
+
+if (USE_PYTHON)
+  message("writing file: " ${CMAKE_CURRENT_BINARY_DIR}/source/boost/python-config.jam)
+  file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/source/boost/python-config.jam)
+  if (USE_PYTHON2)
+    find_package(Python2 REQUIRED)
+    message("python: " ${Python2_EXECUTABLE})
+    file(
+      APPEND ${CMAKE_CURRENT_BINARY_DIR}/source/boost/python-config.jam
+      "using python : ${Python2_VERSION_MAJOR}.${Python2_VERSION_MINOR} : \"${Python2_EXECUTABLE}\" ; \n"
+    )
+    list(APPEND BOOST_PYTHON_VERSIONS "${Python2_VERSION_MAJOR}.${Python2_VERSION_MINOR}")
+  else ()
+  find_package(Python3 REQUIRED)
+    message("python: " ${Python3_EXECUTABLE})
+    file(
+      APPEND ${CMAKE_CURRENT_BINARY_DIR}/source/boost/python-config.jam
+      "using python : ${Python3_VERSION_MAJOR}.${Python3_VERSION_MINOR} : \"${Python3_EXECUTABLE}\" ; \n"
+    )
+    list(APPEND BOOST_PYTHON_VERSIONS "${Python3_VERSION_MAJOR}.${Python3_VERSION_MINOR}")
+  endif()
+  list(JOIN BOOST_PYTHON_VERSIONS "," BOOST_PYTHON_VERSIONS)
+
+  set(BOOST_BUILD_COMMAND ${BOOST_BUILD_COMMAND}  --user-config=${CMAKE_CURRENT_BINARY_DIR}/source/boost/python-config.jam python=${BOOST_PYTHON_VERSIONS} --with-python)
+
+endif()
+
+ExternalProject_Add (
+	${EP_BOOST}
+
+	PREFIX         ${EP_BOOST}/source/boost
+	BUILD_IN_SOURCE 1
+	URL https://downloads.sourceforge.net/project/boost/boost/1.67.0/boost_1_67_0.tar.gz
```

Comment:
why did you change the boost version?  Note that I had it split between version 1.61 and 1.70 for python2 vs python3 as I ran into issues previously with python3 using the older boost and vise versa.  Did you try the build with both python3 and python2 with this version?  I tried it with python2 and 1.67 and it does work on my ubuntu build, though I'm not sure about python3.

Reply:
This version was the only one of the three where I was able to build 'boost_python' for Python2 and Python3 at the same time without any problems.

Reply:
ok great, I had to play version roulette to find one that worked for each.